synapseclient/doze.py

```diff
@@ -0,0 +1,29 @@
+'''
+Created on Sep 21, 2017
+
+@author: bhoff
+
+sleep while checking registered listeners
+'''
+import time
+
+listeners=[]
+
+def add_listener(listener):
```

Comment:
add a safety check so we dont end up calling` listener()` on a non callable variable during `doze`
```
if not callable(listener):
    raise ValueError("listener is not callable")
```